cleanlab/filter.py

```diff
@@ -446,7 +446,11 @@ class 0, 1, ..., K-1.
     if filter_by not in ["low_self_confidence", "low_normalized_margin"]:
         # Remove label issues if given label == model prediction if issues haven't been removed yet
         pred = pred_probs.argmax(axis=1)
-        label_issues_mask[pred == labels] = False
+        mask = pred == labels
+        if K == 2:
+            # Set mask to False wherever pred_probs == 0.5 where num_classes == 2
+            mask = mask & (pred_probs[:, 0] != 0.5)
+        label_issues_mask[mask] = False
 
```

Comment:
Don't forget any updates to find_label_issues from now on must also be considered for find_label_issues_batched

Reply:
Note this actually also happening in this method: https://github.com/cleanlab/cleanlab/blob/master/cleanlab/count.py#L153

I'm worried it is too unsafe to just do this update here, we are likely to miss other related methods where it needs to happen too

Reply:
I've added support in all places I could find